[CORE-8848] `rptest`: adjust compaction settings in `datalake/compaction_test`

WillemKauf opened the pull request:
[Because we now schedule adjacent segment compaction after sliding window compaction](https://redpandadata.atlassian.net/browse/CORE-8848), this test was having trouble trying to reach the desired number of segments while producing.

Increase the timeout as well as the `log_compaction_interval_ms` to allow the test to reach the desired number of segments.


## Backports Required

- [X] none - not a bug fix
- [ ] none - this is a backport
- [ ] none - issue does not exist in previous branches
- [ ] none - papercut/not impactful enough to backport
- [ ] v24.3.x
- [ ] v24.2.x
- [ ] v24.1.x

## Release Notes

* none

WillemKauf committed:
`rptest`: adjust compaction settings in `datalake/compaction_test`

Because we now schedule adjacent segment compaction after sliding
window compaction, this test was having trouble trying to reach
the desired number of segments while producing.

Increase the timeout as well as the `log_compaction_interval_ms`
to allow the test to reach the desired number of segments.

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/61801-->
<!--cbfba68e-ffaa-4abd-b558-95a78165f56f-->
<!--968356aa-0488-4176-9064-1c93f8e221bf-->
<!--title-->
## Retry command for [Build#61801](https://buildkite.com/redpanda/redpanda/builds/61801)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```

/ci-repeat 1
tests/rptest/tests/datalake/compaction_test.py::CompactionGapsTest.test_translation_no_gaps@{"catalog_type":"rest_jdbc","cloud_storage_type":1}
tests/rptest/tests/datalake/compaction_test.py::CompactionGapsTest.test_translation_no_gaps@{"catalog_type":"rest_hadoop","cloud_storage_type":1}
```
<!--/body-->

vbotbuildovich commented:
<!--TEST RESULTS-->
<!--ce976b7c-922c-4c57-93e9-ff0abce835f5-->
<!--TEST RESULTS-->
<!--8d6c1644-2cc4-4168-9c69-e0d98430ab52-->
<!--TEST RESULTS-->
<!--be80b04b-8ce9-4453-8a55-848a073a85de-->
<!--TEST RESULTS-->
<!--7ce576c7-0569-4145-ba73-da2935f07264-->
<!--title-->
## CI test results
<!--/title-->
<!--body-->
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/61801">build#61801</a></summary>


| test_id | test_kind | job_url | test_status | passed |
|:------------------------------------------------------------------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|
| rptest.tests.compaction_recovery_test.CompactionRecoveryTest.test_index_recovery | ducktape | https://buildkite.com/redpanda/redpanda/builds/61801#0194f654-e8be-45a4-bed8-ee01e2d5360c | FLAKY | 1/3 |
| rptest.tests.compaction_recovery_test.CompactionRecoveryTest.test_index_recovery | ducktape | https://buildkite.com/redpanda/redpanda/builds/61801#0194f659-106a-4604-aea7-d4bd3226eda9 | FLAKY | 1/2 |
| rptest.tests.compaction_recovery_test.CompactionRecoveryUpgradeTest.test_index_recovery_after_upgrade | ducktape | https://buildkite.com/redpanda/redpanda/builds/61801#0194f654-e8bf-4294-9f32-b1dda3a2200c | FLAKY | 1/2 |
| rptest.tests.datalake.compaction_test.CompactionGapsTest.test_translation_no_gaps.cloud_storage_type=CloudStorageType.S3.catalog_type=CatalogType.REST_HADOOP | ducktape | https://buildkite.com/redpanda/redpanda/builds/61801#0194f659-106a-4604-aea7-d4bd3226eda9 | FAIL | 0/20 |
| rptest.tests.datalake.compaction_test.CompactionGapsTest.test_translation_no_gaps.cloud_storage_type=CloudStorageType.S3.catalog_type=CatalogType.REST_JDBC | ducktape | https://buildkite.com/redpanda/redpanda/builds/61801#0194f659-1067-43af-bea3-8ac5c24324ae | FAIL | 0/20 |
| rptest.tests.e2e_shadow_indexing_test.EndToEndShadowIndexingTestWithDisruptions.test_write_with_node_failures.cloud_storage_type=CloudStorageType.ABS | ducktape | https://buildkite.com/redpanda/redpanda/builds/61801#0194f654-e8bf-4294-9f32-b1dda3a2200c | FLAKY | 1/2 |
| rptest.tests.e2e_shadow_indexing_test.ShadowIndexingWhileBusyTest.test_create_or_delete_topics_while_busy.short_retention=True.cloud_storage_type=CloudStorageType.ABS | ducktape | https://buildkite.com/redpanda/redpanda/builds/61801#0194f659-106a-4604-aea7-d4bd3226eda9 | FLAKY | 1/2 |
| rptest.tests.log_compaction_test.LogCompactionTest.compaction_stress_test.cleanup_policy=compact.delete.key_set_cardinality=1000.storage_compaction_key_map_memory_kb=3 | ducktape | https://buildkite.com/redpanda/redpanda/builds/61801#0194f659-1069-48f3-84cb-b3b611125f0e | FLAKY | 1/2 |
| rptest.tests.log_compaction_test.LogCompactionTest.compaction_stress_test.cleanup_policy=compact.key_set_cardinality=1000.storage_compaction_key_map_memory_kb=3 | ducktape | https://buildkite.com/redpanda/redpanda/builds/61801#0194f659-106a-4604-aea7-d4bd3226eda9 | FLAKY | 1/2 |
| rptest.tests.simple_e2e_test.SimpleEndToEndTest.test_consumer_interruption | ducktape | https://buildkite.com/redpanda/redpanda/builds/61801#0194f654-e8bc-47a1-adbf-12c11d80ba67 | FLAKY | 1/2 |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/61807">build#61807</a></summary>


| test_id | test_kind | job_url | test_status | passed |
|:------------------------------------------------------------------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|
| rptest.tests.compaction_recovery_test.CompactionRecoveryTest.test_index_recovery | ducktape | https://buildkite.com/redpanda/redpanda/builds/61807#0194f77b-dd54-4034-90ce-1dcaaf816bcb | FLAKY | 1/3 |
| rptest.tests.compaction_recovery_test.CompactionRecoveryUpgradeTest.test_index_recovery_after_upgrade | ducktape | https://buildkite.com/redpanda/redpanda/builds/61807#0194f769-f0ba-4a14-8fdd-cadb456ac157 | FLAKY | 1/2 |
| rptest.tests.log_compaction_test.LogCompactionTest.compaction_stress_test.cleanup_policy=compact.delete.key_set_cardinality=1000.storage_compaction_key_map_memory_kb=3 | ducktape | https://buildkite.com/redpanda/redpanda/builds/61807#0194f769-f0bb-4508-8f27-faf0ea4b5bba | FLAKY | 1/2 |
| rptest.tests.log_compaction_test.LogCompactionTest.compaction_stress_test.cleanup_policy=compact.key_set_cardinality=1000.storage_compaction_key_map_memory_kb=3 | ducktape | https://buildkite.com/redpanda/redpanda/builds/61807#0194f769-f0bd-44b4-a94b-ba347f321834 | FLAKY | 1/3 |
| rptest.tests.scaling_up_test.ScalingUpTest.test_scaling_up_with_recovered_topic | ducktape | https://buildkite.com/redpanda/redpanda/builds/61807#0194f769-f0bd-44b4-a94b-ba347f321834 | FLAKY | 1/2 |
| storage_single_thread_rpunit.storage_single_thread_rpunit | unit | https://buildkite.com/redpanda/redpanda/builds/61807#0194f71f-d9d2-431e-a157-d1fde5eb99e2 | FLAKY | 1/2 |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/61813">build#61813</a></summary>


| test_id | test_kind | job_url | test_status | passed |
|:------------------------------------------------------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|
| rptest.tests.datalake.compaction_test.CompactionGapsTest.test_translation_no_gaps.cloud_storage_type=CloudStorageType.S3.catalog_type=CatalogType.REST_JDBC | ducktape | https://buildkite.com/redpanda/redpanda/builds/61813#0194f854-3264-4611-a8d4-835f5f966905 | FLAKY | 30/34 |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/61823">build#61823</a></summary>


| test_id                                                                                                                                                                 | test_kind   | job_url                                                                                   | test_status   | passed   |
|:------------------------------------------------------------------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|
| rptest.tests.compaction_recovery_test.CompactionRecoveryTest.test_index_recovery                                                                                        | ducktape    | https://buildkite.com/redpanda/redpanda/builds/61823#0194fb49-07da-4a40-979d-ca11b54e3016 | FLAKY         | 1/2      |
| rptest.tests.compaction_recovery_test.CompactionRecoveryUpgradeTest.test_index_recovery_after_upgrade                                                                   | ducktape    | https://buildkite.com/redpanda/redpanda/builds/61823#0194fb49-07dc-4259-b282-391c8dedf5a9 | FLAKY         | 1/2      |
| rptest.tests.log_compaction_test.LogCompactionTest.compaction_stress_test.cleanup_policy=compact.delete.key_set_cardinality=1000.storage_compaction_key_map_memory_kb=3 | ducktape    | https://buildkite.com/redpanda/redpanda/builds/61823#0194fb41-ef78-4974-b373-7441153ddcb3 | FLAKY         | 1/2      |
| rptest.tests.log_compaction_test.LogCompactionTest.compaction_stress_test.cleanup_policy=compact.key_set_cardinality=1000.storage_compaction_key_map_memory_kb=10       | ducktape    | https://buildkite.com/redpanda/redpanda/builds/61823#0194fb41-ef78-4974-b373-7441153ddcb3 | FLAKY         | 1/2      |
</details>
<!--/body-->

bharathv commented:
/ci-repeat 1
skip-build
skip-units
dt-repeat=30
tests/rptest/tests/datalake/compaction_test.py

WillemKauf commented:
@bharathv the timeout issues in `CompactionGapsTest` seem fixed by this PR, but there are still some failures such as

```
ducktape.cluster.remoteaccount.RemoteCommandError: root@docker-rp-21: Command 'python3 /tmp/compute_storage.py --data-dir=/var/lib/redpanda/data' returned non-zero exit status 1. Remote error message: b'Traceback (most recent call last):\n  File "/tmp/compute_storage.py", line 196, in <module>\n    output = compute_size(data_dir, args.sizes, args.md5, args.print_flat)\n  File "/tmp/compute_storage.py", line 151, in compute_size\n    for partition in safe_listdir(topic):\n  File "/tmp/compute_storage.py", line 89, in safe_listdir\n    return [f for f in p.iterdir()]\n  File "/tmp/compute_storage.py", line 89, in <listcomp>\n    return [f for f in p.iterdir()]\n  File "/usr/lib/python3.10/pathlib.py", line 1017, in iterdir\n    for name in self._accessor.listdir(self):\nNotADirectoryError: [Errno 20] Not a directory: \'/var/lib/redpanda/data/datalake_staging/0-2c771258-a4b6-4750-870b-238fa1da198f.parquet\'\n'
```

Separate issue?

WillemKauf commented:
I think [compute_storage.py](https://github.com/redpanda-data/redpanda/blob/01a9c7b01254f603a30fef599417d3b20e9c18bf/tests/rptest/remote_scripts/compute_storage.py#L82-L91) should look like

```python
try:                                                                                                                  
  return [f for f in p.iterdir()]                                                                                   
except FileNotFoundError:                                                                                             
   return []                                                                                                         
except NotADirectoryError:                                                                                            
  return []  
```

but I'm not sure yet, quite tired

WillemKauf committed:
`rptest`: add sanity check to `safe_listdir()`

WillemKauf commented:
Force push to:
* add sanity check to `compute_storage.py` in `safe_listdir()`

WillemKauf commented on tests/rptest/remote_scripts/compute_storage.py:
```diff
@@ -69,7 +69,7 @@ def __iter__(self) -> Iterator[Header]:
 def safe_isdir(p: Path) -> bool:
     """
-    It's valid for files to be deleted at any time, 
+    It's valid for files to be deleted at any time,
```
trailing whitespace removal

WillemKauf commented on tests/rptest/remote_scripts/compute_storage.py:
```diff
@@ -81,10 +81,13 @@ def safe_isdir(p: Path) -> bool:
 def safe_listdir(p: Path) -> list[Path]:
     """
-    It's valid for directories to be deleted at any time, 
+    It's valid for directories to be deleted at any time,
```
trailing whitespace removal

bharathv commented:
I think we should also add "datalake_staging" to this [list here](https://github.com/redpanda-data/redpanda/blob/01a9c7b01254f603a30fef599417d3b20e9c18bf/tests/rptest/remote_scripts/compute_storage.py#L146), it was recently moved under the data directory.

WillemKauf committed:
`rptest`: consider `datalake_staging` in `compute_size()`

The `datalake_staging` folder was recently moved under the redpanda
data directory. It should not be considered as a namespace in `compute_size()`.

WillemKauf commented:
>I think we should also add "datalake_staging" to this [list here](https://github.com/redpanda-data/redpanda/blob/01a9c7b01254f603a30fef599417d3b20e9c18bf/tests/rptest/remote_scripts/compute_storage.py#L146), it was recently moved under the data directory.

Awesome, added.
